src/components/Popover/Popover.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react'
+import ReactDOM, { findDOMNode } from 'react-dom'
+import PropTypes from 'prop-types'
+import Popper from 'popper.js'
+
+class Popover extends React.Component {
+  componentDidMount() {
+    this.initPopper()
+  }
+
+  componentWillUnmount() {
+    this.destroyPopper()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      prevProps.visible !== this.props.visible ||
+      prevProps.placement !== this.props.placement ||
+      prevProps.children !== this.props.children
+    ) {
+      this.destroyPopper()
+      this.initPopper()
+    }
+  }
+
+  initPopper = () => {
+    const { openerRef, placement, gutter } = this.props
+    if (placement && !this.popper) {
+      this.popper = new Popper(openerRef, findDOMNode(this), {
+        placement,
+        modifiers: {
+          offset: { offset: `0, ${gutter}` },
+        },
+      })
+    }
+  }
+
+  destroyPopper = () => {
+    if (this.props.placement && this.popper) {
+      this.popper.destroy()
+      this.popper = undefined
+    }
+  }
+
+  render() {
+    const { top, left, zIndex, children, containerRef } = this.props
+    return ReactDOM.createPortal(
+      <div style={{ position: 'absolute', top, left, zIndex }}>{children}</div>,
+      containerRef
+    )
+  }
+}
+
+const { Consumer, Provider } = React.createContext('Popover')
+
+class ContainerProvider extends React.Component {
+  state = { el: null }
+  handleRef = el => {
+    this.setState({ el })
+  }
+  render() {
+    const { el } = this.state
+    const { children } = this.props
+    return (
+      <Provider value={el}>
+        <div ref={this.handleRef}>{children}</div>
+      </Provider>
+    )
+  }
+}
+
+class PopoverWithProvider extends React.Component {
+  static propTypes = {
+    openerRef: PropTypes.node,
```

Comment:
`PropTypes.node` is to check React nodes, but we want to check for a (required) DOM node here:

```suggestion
    openerRef: PropTypes.instanceOf(Element).isRequired,
```

Reply:
It's not required when doing positioning relative to the mouse

Reply:
OpenerRef is not always required..

Reply:
Made it required anyways, since I ran into a problem that requires it.